### DISCUSSION TITLE
Install pkgs globally by defining them in package.json

### DISCUSSION THREAD

**Discussion by ghost** [npm/feedback#181] [Registry]
This would be great for installing pkgs globally so we can use the cli for it.

**Comment by ghost** [comment]
Perfect if we have an open source project and anyone installing it can install globally pkgs like nodemon or something important 😏

**Comment by ljharb** [comment]
Are you asking for a package.json for global modules - so names/versions can be persistent? Or are you asking for something else?

**Comment by ghost** [comment]
@ljharb just as we have `devDependencies` in package.json, we can have `globalDependencies` which will automatically install them as `-g` instead of `-d`

**Reply by ljharb** [reply]
Global installs are a bad practice; why would we want that?

**Reply by ghost** [reply]
@ljharb probably for cli applications like "nodemon"

**Comment by ghost** [comment]
It's a way of installing global modules automatically from package.json

**Reply by MylesBorins** [reply]
All packages that are installed that have binary's are linked in the package.scripts.

everything ends up in node_modules/.bin and can be referred to by their binary name in any script.

You can even add that to your path and automatically get any installed modules available in your path.

https://github.com/MylesBorins/dot-files/blob/main/.bash_profile#L10-L11

I don't think it would be a good idea to have the installation of a project change the global system

**Reply by ghost** [reply]
Hmm, but what about cli applications?

**Comment by ghost** [comment]
Or atleast we can do `npm link` for global dependencies listed in package.json

**Comment by ghost** [comment]
Ok, so I'm making an npm package to do the job, so if anyone wants to use them, they can.
